### PR TITLE
fix(SCT-1082): reduce json size/tidy up controllers

### PR DESCRIPTION
**Pull request by JohnFarrellDev**
## Link to JIRA ticket

[Ticket](https://hackney.atlassian.net/browse/SCT-1082) (Though it potentially helps speed up the response from every single API endpoint)

### *What is the problem we're trying to solve*

We want to reduce response size to increase speed of our API call to GET /api/vi/submissions
Current response includes a property even when the value is null, we can just not include the property or null value in the response and reduce the JSON response size.

Screenshot from 2021-08-23 08-43-30

### *What changes have we introduced*

Updated the JSON serializer to remove the property if the value is null from our JSON response.

#### _Checklist_

- [ ] Added end-to-end (i.e. integration) tests where necessary e.g to test complete functionality of a newly added feature
- [ ] Added tests to cover all new production code
- [ ] Added comments to the README or updated relevant documentation _(add link to documentation)_, where necessary.
- [ ] Checked all code for possible refactoring
- [ ] Code pipeline builds correctly

### *Follow up actions after merging PR*

Are there any next steps that need to addressed after merging this PR? Add them here.

### CODE REVIEW

File: SocialCareCaseViewerApi/Startup.cs [hunk 1]
```diff
@@ -43,7 +34,11 @@ public static void ConfigureServices(IServiceCollection services)
             services
                 .AddMvc()
                 .SetCompatibilityVersion(CompatibilityVersion.Version_3_0)
-                .AddFluentValidation();
+                .AddFluentValidation()
+                .AddJsonOptions(options =>
+                {
+                    options.JsonSerializerOptions.IgnoreNullValues = true;
+                });
```
Comment (JohnFarrellDev) [inline SocialCareCaseViewerApi/Startup.cs:41]: This is the change to make null properties not be returned in our JSON response

Comment (Renny4Real) [reply SocialCareCaseViewerApi/Startup.cs:41]: Would the front end need to be aware of which properties are not going to be returned. Would it be expecting a specific object and somehow know what to do if a specific property does not exist?

File: SocialCareCaseViewerApi/Startup.cs [hunk 2]
```diff
@@ -116,22 +111,9 @@ public static void ConfigureServices(IServiceCollection services)
...
-            services.AddTransient<IValidator<PatchWarningNoteRequest>, PatchWarningNoteRequestValidator>();
-            services.AddTransient<IValidator<CreateTeamRequest>, CreateTeamRequestValidator>();
-            services.AddTransient<IValidator<GetTeamsRequest>, GetTeamsRequestValidator>();
-            services.AddTransient<IValidator<CreateCaseSubmissionRequest>, CreateCaseSubmissionRequestValidator>();
-            services.AddTransient<IValidator<UpdateCaseSubmissionRequest>, UpdateCaseSubmissionRequestValidator>();
-            services
-                .AddTransient<IValidator<UpdateFormSubmissionAnswersRequest>, UpdateFormSubmissionAnswersValidator>();
-            services.AddTransient<IValidator<CreatePersonalRelationshipRequest>, CreatePersonalRelationshipRequestValidator>();
-            services.AddTransient<IValidator<QueryCaseSubmissionsRequest>, QueryCaseSubmissionsValidator>();
-
```
Comment (Renny4Real) [inline SocialCareCaseViewerApi/Startup.cs:134]: Why are we removing these? Do we no longer need the validators?

File: SocialCareCaseViewerApi/V1/Controllers/BaseController.cs [hunk 3]
```diff
@@ -1,29 +0,0 @@
...
-                var settings = new JsonSerializerSettings();
-                settings.Formatting = Formatting.Indented;
-                settings.ContractResolver = new CamelCasePropertyNamesContractResolver();
-
-                settings.DateTimeZoneHandling = DateTimeZoneHandling.Utc;
-                settings.DateFormatHandling = DateFormatHandling.IsoDateFormat;
-
-                return settings;
-            };
-        }
```
Comment (Renny4Real) [inline SocialCareCaseViewerApi/V1/Controllers/BaseController.cs:27]: Would changing the super class of our constructors from `BaseController` to `Controller` mean we might lose these JSON settings? Would we want these settings?

**Review by Renny4Real — COMMENTED** [review]
Looks good. Just a few comments

File: SocialCareCaseViewerApi/V1/Controllers/BaseController.cs (hunk 3 above)
Comment (JohnFarrellDev) [reply SocialCareCaseViewerApi/V1/Controllers/BaseController.cs:27]: As far as I could tell these settings weren't actually implemented.

Initially I tried adding the removal of the null properties here and it had 0 impact on the response.

File: SocialCareCaseViewerApi/Startup.cs (hunk 2 above)
Comment (JohnFarrellDev) [reply SocialCareCaseViewerApi/Startup.cs:134]: I don't believe these are doing anything.

In our constructors we actually the validator and call it against the request there doesn't seem to be a requirement to inject the validator into the Startup class.

**Pull request merged**